### PR TITLE
CMake modifications inspired by work on Fugaku

### PR DESCRIPTION
**Pull request by benson31**
While these are driven by some of the issues we had on Fugaku, I believe they're relevant to `develop`.

### CODE REVIEW

**Review by timmoon10 — APPROVED** [review]
LGTM

**Comment by benson31** [comment]
Bamboo is happy enough for both CPU and GPU systems.

File: CMakeLists.txt [hunk 1]
```diff
@@ -424,7 +397,8 @@ endif ()
 # TODO: This probably shouldn't be the case and should disable specific
 # callbacks when OpenCV is not available
 if (LBANN_WITH_TBINF AND NOT LBANN_HAS_OPENCV)
-  message(FATAL_ERROR "TensorBoard requires OpenCV. Enable LBANN_WITH_VISION to use TensorBoard.")
+  message(FATAL_ERROR "TensorBoard requires OpenCV. "
```
Comment (mrwyattii) [inline CMakeLists.txt:400]: Is this check necessary?  `LBANN_WITH_TBINF` is set to `LBANN_WITH_VISION` (line 135), so this *should* never be triggered

**Review by mrwyattii — APPROVED** [review]
Left one comment/suggestion, but overall looks good

File: CMakeLists.txt (hunk 1 above)
Comment (benson31) [reply CMakeLists.txt:400]: That's the default value only. You might manually say `LBANN_WITH_VISION=ON`, `LBANN_WITH_TBINF=OFF` (this should be fine) but you could just as easily say: `LBANN_WITH_VISION=OFF`, `LBANN_WITH_TBINF=ON`. This is the case we're protecting against.

**Review by bvanessen — APPROVED** [review]

**Pull request merged**